packages/tux-adapter-contentful/src/query-api.ts

```diff
@@ -93,37 +91,32 @@ class QueryApi {
     }
 
     // Add included models to items
-    for (const item of result.items) {
-      this.linkFields(item, linkMap)
+    for (let i = 0; i < result.items.length; i++) {
+      const item = result.items[i]
+      result.items[i] = this.linkEntry(item, linkMap)
     }
   }
 
-  private linkFields(item: any, linkMap: LinkMap) {
-    if (!item) {
-      return
+  private linkEntry(item: any, linkMap: LinkMap): any {
+    const hasSys = !!item.sys
+    const isLink = hasSys && item.sys.type === 'Link'
+    const isArray = item instanceof Array
+
+    if (item.fields) {
+      Object.keys(item.fields).forEach(fieldName => {
+        item.fields[fieldName] = this.linkEntry(item.fields[fieldName], linkMap)
+      })
```

Comment:
Also, if `linkEntry` can handle arrays, you could call it directly like `result.items = linkEntries(result.items, linkMap)`